File: src/main/java/com/kigya/patterns/java/decorator/decorators/Mocha.java
package com.kigya.patterns.java.decorator.decorators;

import com.kigya.patterns.java.decorator.component.Beverage;

public class Mocha extends CondimentDecorator {
    public Mocha(Beverage beverage) {
        this.beverage = beverage;
    }

    public String getDescription() {
        return beverage.getDescription() + ", Mocha";
    }

    public double cost() {
        return .20 + beverage.cost();
    }
}

File: src/main/java/com/kigya/patterns/kotlin/decorator/decorators/Mocha.kt
package com.kigya.patterns.kotlin.decorator.decorators

import com.kigya.patterns.kotlin.decorator.component.Beverage

class Mocha(private val beverage: Beverage) : CondimentDecorator() {

    override val description: String
        get() = "${beverage.description}, Mocha"

    override fun cost(): Double = beverage.cost() + .20
}